framework/caching/ArrayCache.php

```diff
@@ -84,6 +104,26 @@ protected function deleteValue($key)
     protected function flushValues()
     {
         $this->_cache = [];
+
+        return true;
+    }
+
+
+    /**
+     * Removes the expired data values.
+     * @param bool $force whether to enforce the garbage collection regardless of [[gcProbability]].
+     * Defaults to false, meaning the actual deletion happens with the probability as specified by [[gcProbability]].
+     */
+    public function gc($force = false)
+    {
+        if ($force || mt_rand(0, 1000000) < $this->gcProbability){
+            foreach($this->_cache as $key => $item){
+                if (!isset($item[0], $item[1]) || (0 !== $item[0] && $item[1] < microtime(true))){
+                    $this->deleteValue($key);
```

Comment:
You have potential infinite recursion here: `gc()` calls `deleteValue()` which calls `gc()`.

Reply:
Has been fixed